src/rgw/rgw_asio_client.cc

```diff
@@ -47,22 +47,22 @@ int ClientIO::init_env(CephContext *cct)
       continue;
     }
 
-    static const std::string_view HTTP_{"HTTP_"};
-
-    char buf[name.size() + HTTP_.size() + 1];
-    auto dest = std::copy(std::begin(HTTP_), std::end(HTTP_), buf);
-    for (auto src = name.begin(); src != name.end(); ++src, ++dest) {
-      if (*src == '-') {
-        *dest = '_';
-      } else if (*src == '_') {
-        *dest = '-';
-      } else {
-        *dest = std::toupper(*src);
-      }
-    }
-    *dest = '\0';
-
-    env.set(buf, std::string(value));
+    static constexpr std::string_view HTTP_{"HTTP_"};
+
+    std::string key{HTTP_};
+    key.reserve(name.size() + HTTP_.size());
+    std::transform(name.cbegin(), name.cend(),
+		   std::back_inserter(key),
+		   [](char c) -> char {
```

Comment:
This looks fine, but don't we have this very function ensconced in our utilities? I think I've seen it a few times. (I see we do it again, below.)

Reply:
So factored.